src/3rdparty/squirrel/squirrel/squtils.h

```diff
@@ -90,7 +90,7 @@ template<typename T> class sqvector
 	{
 		_vals[idx].~T();
 		if(idx < (_size - 1)) {
-			memmove(&_vals[idx], &_vals[idx+1], sizeof(T) * (_size - (size_t)idx - 1));
+			memmove(static_cast<void *>(&_vals[idx]), &_vals[idx+1], sizeof(T) * (_size - (size_t)idx - 1));
```

Comment:
If T is not a POD type this is illegal. (I don't know if T can be non-POD types in the uses of this type in Squirrel.) The line should be replaced with a loop that uses `std::move` to move the elements down individually, and trust that an optimizing compiler will recognize the loop as a move-down operation.

Reply:
That'd be preferable, except that std::move is C++11 only, and we haven't quite made that switch yet. The alternative is std::copy... but that's not a move. memmove has worked fine so far, so we'll have to assume it's valid for the types it's used on

Reply:
The other alternative would of course be to just use assignment and hope `operator =` doesn't cause problems for the classes involved.